### ISSUE TITLE
Understanding when vmagent rejects scrapes larger than `maxScrapeSize`

### ISSUE THREAD

**Issue by AeroNotix** [VictoriaMetrics/VictoriaMetrics#1639]
Hi,

I just had an issue where metrics seemingly stopped coming in, eventually traced back to kube-state-metrics just tipping over the threshold we have set for `maxScrapeSize`.

Docs on `maxScrapeSize`

```
  -promscrape.maxScrapeSize size
    	The maximum size of scrape response in bytes to process from Prometheus targets. Bigger responses are rejected
    	Supports the following optional suffixes for size values: KB, MB, GB, KiB, MiB, GiB (default 16777216)
```

Looking through documentation I am not seeing if there is any mention of any metrics on vmagent itself when a scrape is rejected.

My question is, does that exist? Does vmagent increment any counters (with appropriate labels) when scrapes are being rejected?

If not, is that something that your team would be willing to include? I am willing to do the work, if so.

**Comment by AeroNotix** [comment]
Looking at the source, it appears this only logs:

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/728c4c3841bc791040a97a84fd2e20686a78b433/lib/promscrape/client.go#L340-L346

**Comment by valyala** [comment]
@AeroNotix , could you look at the commit a0313c046bdaee45a3dfd8a5f449ccbc30dd88fa , which adds the `vm_promscrape_max_scrape_size_exceeded_errors_total` metric? If you feel it could be improved, then please send a pull request.

**Comment by AeroNotix** [comment]
This is immediately useful, thank you.

**Comment by valyala** [comment]
FYI, the `vm_promscrape_max_scrape_size_exceeded_errors_total` metric is available starting from [v1.66.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by AeroNotix** [comment]
Appreciate it! Will update and start alerting on this metric. Thanks again.
